diff --git a/BUILD.bazel b/BUILD.bazel
index 8dd76e07..9e8e284 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -34,1 +34,4 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+load("@rules_python//python:defs.bzl", "py_binary")
+
 package(
@@ -66,7 +69,7 @@ genrule(
 genrule(
     name = "gen_extension_headers_h",
-    srcs = ["glslang/ExtensionHeaders", "gen_extension_headers.py"],
+    srcs = ["gen_extension_headers.py"],
     outs = ["glslang/glsl_intrinsic_header.h"],
-    cmd_bash = "$(location gen_extension_headers) -i  $(location glslang/ExtensionHeaders) -o $(location glslang/glsl_intrinsic_header.h)",
+    cmd_bash = "$(location gen_extension_headers) -i  glslang/ExtensionHeaders -o $(location glslang/glsl_intrinsic_header.h)",
     tools = [":gen_extension_headers"],
 )
